addons/mail/static/src/components/message_list/message_list.js

```diff
@@ -331,15 +354,13 @@ export class MessageList extends Component {
      */
     _adjustScrollForExtraMessagesAtTheEnd() {
         const {
-            hasAutoScrollOnMessageReceived,
             hasScrollAdjust,
-            order,
         } = this._lastRenderedValues();
         if (!this._getScrollableElement() || !hasScrollAdjust) {
             return;
         }
-        if (!hasAutoScrollOnMessageReceived) {
-            if (order === 'desc' && this._willPatchSnapshot) {
+        if (!this.threadView.hasAutoScrollOnMessageReceived) {
+            if (this.threadView.order === 'desc' && this._willPatchSnapshot) {
```

Comment:
better take values from _lastRenderedValues to avoid async issues, there should be no access to threadView from this